cpp/include/tensorrt_llm/executor/executor.h

```diff
@@ -1420,14 +1420,29 @@ class LogitsPostProcessorConfig
 class CacheTransceiverConfig
 {
 public:
-    explicit CacheTransceiverConfig(std::optional<size_t> maxNumTokens = std::nullopt);
+    enum class CommType : std::uint8_t
+    {
+        UNKNOWN = 0,
+        MPI = 1,
+        UCX = 2,
+        NIXL = 3
+    };
+    explicit CacheTransceiverConfig(bool enableCacheTransceiver = false,
+        std::optional<CommType> commType = std::nullopt, std::optional<size_t> maxNumTokens = std::nullopt);
 
     bool operator==(CacheTransceiverConfig const& other) const;
 
     [[nodiscard]] std::optional<size_t> getMaxNumTokens() const;
     void setMaxNumTokens(size_t maxNumTokens);
 
+    void setEnableCacheTransceiver(bool enableCacheTransceiver);
```

Comment:
I would suggest removing the two `set` functions and instead using the constructor to initialize the members, in order to maintain the [class invariant](https://en.wikipedia.org/wiki/Class_invariant). The concept of invariants is a key distinction between `class` and `struct`, and I believe we should adhere to it whenever possible.